shapely/_geometry_helpers.pyx

```diff
@@ -292,7 +292,7 @@ cdef void _deallocate_arr(void* handle, np.intp_t[:] arr, Py_ssize_t last_geom_i
     for i in range(last_geom_i):
         g = <GEOSGeometry *>arr[i]
         if g != NULL:
-            GEOSGeom_destroy_r(handle, <GEOSGeometry *>arr[i])
+            GEOSGeom_destroy_r(<GEOSContextHandle_t> handle, <GEOSGeometry *>arr[i])
```

Comment:
Actually, could you change the `void* handle` in `GEOSContextHandle_t handle` in the `_deallocate_arr` signature a few lines above? That should also fix it and feels better (everywhere we use this helper function, the handler is declared to be a context handle)

Reply:
Sure. I'll test it on my local code and do that if it works, but it might be Monday before I get to it

Reply:
(small ping, otherwise I can also push myself after the weekend)

Reply:
@schwehr updated here (I want to get this out in a 2.1.1 release), going to assume my suggestion is also fixing the original issue you noticed (but please test and let us know if not!)